docs/enhancements/dynamic-fetching-of-openshift-versions.md

```diff
@@ -0,0 +1,280 @@
+---
+title: dynamic-fetching-of-openshift-versions
+authors:
+  - "@osherdp"
+creation-date: 2023-01-12
+last-updated: 2023-01-12
+---
+
+# Dynamic Fetching of OpenShift Versions
+
+Letting users of Assisted Installer API and UI to choose an OpenShift version
+out of a variety of supported versions, all without the need to specifically
+configure each of those versions.
+
+## Summary
+
+Currently, we have defined processes to support the growing list of OpenShift
+versions. It can be separated to the following cases:
+* Adding a newly released minor version of OpenShift when it's first introduced
+  as part of an EC (Engineering Candidate), FC (Feature Complete) or RC (Release
+  Complete) versions. For example, OpenShift 4.12 had its first appearance as
+  version ``4.12.0-ec.0``. With the new version we usually also add live-iso
+  discovery images that are appropriate for the release.
+* Updating beta releases like EC/FC/RC by following candidate channels like
+  ``candidate-4.12``.
+* Updating to the first official stable version when it's GA (Globally
+  Available). For example, updating to 4.12.0 once it's published and available
+  for customers to use.
+* Updating stable releases following stable channels like ``stable-4.12``.
+
+All those actions consist of either manual updates by developers or automatic
+tools like https://gitlab.cee.redhat.com/assisted-installer-ops/assisted-installer-cicd.
+Either manual or automatic, we usually override releases with the new ones (e.g.
+replacing 4.12.0 with 4.12.1) to make sure we don't grow too much our configuration
+files. As per special requests (usually made by customers), we append additional
+versions. For example, see:
```

Comment:
```suggestion
replacing 4.12.0 with 4.12.1) to prevent our configuration files from becoming too
large. We retain specific versions as per special requests (usually made by
customers/partners). For example, see:
```